common-util/src/main/java/org/sunbird/services/sso/impl/KeyCloakRsaKeyFetcher.java

```diff
@@ -0,0 +1,108 @@
+package org.sunbird.services.sso.impl;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.RSAPublicKeySpec;
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.PropertiesCache;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * @author  github.com/iostream04 This class will connect to key cloak server and
+ *         fetch the sso_publicKey using http call.
+ */
+
+public class KeyCloakRsaKeyFetcher {
+
+  public PublicKey getPublicKeyFromKeyCloak(String url, String realm) {
+    try {
+      Decoder urlDecoder = Base64.getUrlDecoder();
+      KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+      Map<String, String> valueMap = getValuesFromJson(requestKeyFromKeycloak(url, realm));
```

Comment:
Will result in null pointer exception if response is null from requestKeyFromKeycloak due to any network error or exception.